cmd/snap/error.go

```diff
@@ -0,0 +1,127 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"go/doc"
+	"os"
+	"os/user"
+	"strings"
+
+	"golang.org/x/crypto/ssh/terminal"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/osutil"
+)
+
+var errorPrefix = i18n.G("error: %v\n")
+
+func termSize() (width, height int) {
+	if f, ok := Stdout.(*os.File); ok {
+		width, height, _ = terminal.GetSize(int(f.Fd()))
+	}
+
+	if width <= 0 {
+		width = int(osutil.GetenvInt64("COLUMNS"))
+	}
+
+	if height <= 0 {
+		height = int(osutil.GetenvInt64("COLUMNS"))
+	}
+
+	if width < 40 {
+		width = 80
+	}
+
+	if height < 15 {
+		height = 25
+	}
+
+	return width, height
+}
+
+func fill(para string) string {
+	width, _ := termSize()
+
+	if width > 100 {
+		width = 100
+	}
+
+	width--
+
+	// 3 is the %v\n, which will be present in any locale
+	indent := len(errorPrefix) - 3
+	var buf bytes.Buffer
+	doc.ToText(&buf, para, strings.Repeat(" ", indent), "", width)
+
+	return strings.TrimSpace(buf.String())
+}
+
+func clientErrorToCmdMessage(snapName string, err *client.Error) (string, error) {
+	isError := true
```

Comment:
maybe also have a mentionsSnap flag? and avoid the returns in the middle of the switch?  I found them a bit confusing but maybe it's just me